[CI] IngestCommonClientYamlTestSuiteIT test {yaml=ingest/310_reroute_processor/Test data stream with lazy rollover obtains pipeline from template} failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #4708 / oraclelinux-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/scszzc2uf3vao)
- [elasticsearch-pull-request #39849 / part-1](https://gradle-enterprise.elastic.co/s/4k6kvogpum64s)
- [elasticsearch-periodic-platform-support #4686 / sles-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/dcn2fbsvh2x6y)
- [elasticsearch-periodic #4696 / concurrent-search-tests](https://gradle-enterprise.elastic.co/s/xry6jnciogiya)

**Reproduction Line:**
```
./gradlew ":modules:ingest-common:yamlRestTest" --tests "org.elasticsearch.ingest.common.IngestCommonClientYamlTestSuiteIT.test {yaml=ingest/310_reroute_processor/Test remove then add reroute processor with and without lazy rollover}" -Dtests.seed=6629E496FCE6A5F8 -Dtests.locale=pt -Dtests.timezone=Australia/West -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.ingest.common.IngestCommonClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Byaml%3Dingest%2F310_reroute_processor%2FTest%20data%20stream%20with%20lazy%20rollover%20obtains%20pipeline%20from%20template%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [ingest/310_reroute_processor:464]: got unexpected warning header [
	299 Elasticsearch-8.17.0-325a3051cbf6f4c7a8cc12540bccf39bc23308a6 "index template [template-1] has index patterns [data-stream-1] matching patterns from existing older templates [global] with patterns (global => [*]); this template [template-1] will take precedence during new index creation"
]

```

**Issue Reasons:**
- [8.x] 4 failures in test test {yaml=ingest/310_reroute_processor/Test data stream with lazy rollover obtains pipeline from template} (1.4% fail rate in 284 executions)
- [8.x] 2 failures in pipeline elasticsearch-periodic-platform-support (22.2% fail rate in 9 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -199,12 +199,6 @@ tests:
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: test {p0=search/380_sort_segments_on_timestamp/Test that index segments are NOT sorted on timestamp field when @timestamp field is dynamically added}
   issue: https://github.com/elastic/elasticsearch/issues/116221
-- class: org.elasticsearch.ingest.common.IngestCommonClientYamlTestSuiteIT
-  method: test {yaml=ingest/310_reroute_processor/Test remove then add reroute processor with and without lazy rollover}
-  issue: https://github.com/elastic/elasticsearch/issues/116158
-- class: org.elasticsearch.ingest.common.IngestCommonClientYamlTestSuiteIT
-  method: test {yaml=ingest/310_reroute_processor/Test data stream with lazy rollover obtains pipeline from template}
-  issue: https://github.com/elastic/elasticsearch/issues/116157
 - class: org.elasticsearch.ingest.geoip.EnterpriseGeoIpDownloaderIT
   method: testEnterpriseDownloaderTask
   issue: https://github.com/elastic/elasticsearch/issues/115163

